### PR TITLE
Add auth support using 3scale custom header

### PR DESCRIPTION
**Pull request by samuzzal-choudhury**

### CODE REVIEW

File: tests/test_token_handling.py [hunk 1]
```diff
@@ -317,8 +326,12 @@ def testing_method():
        side_effect=mocked_fetch_public_keys_4, create=True)
 @patch("fabric8a_auth.auth.get_token_from_auth_header",
        side_effect=mocked_get_token_from_auth_header, create=True)
+@patch.dict(os.environ, {"THREESCALE_ACCOUNT_SECRET": "secret1"})
```
Comment (humaton) [inline tests/test_token_handling.py]: Could you please create new test scenarios for your feature?
The update to tests has disabled they original purpose.

Comment (samuzzal-choudhury) [reply tests/test_token_handling.py]: @humaton please review again. Added new tests. 

FYI - the mock function `_mocked_ret_threescale_account_secret_header_none` is required and the environment variable `THREESCALE_ACCOUNT_SECRET` needs to be set for successfully test the authorization based on osio token.

File: tests/test_token_handling.py [hunk 2]
```diff
@@ -149,6 +149,16 @@ def mocked_get_token_from_auth_header():
     return jwt.encode(payload, PRIVATE_KEY, algorithm='RS256').decode("utf-8")
 
 
+def mocked_ret_threescale_account_secret_header_none():
```
Comment (humaton) [inline tests/test_token_handling.py]: probably typo ret -> get

**Comment by samuzzal-choudhury** [comment]
@humaton thanks for reviewing. I fixed the typo. Will merge this after the build is through.

**Pull request merged**